linqit/__init__.py

```diff
@@ -105,21 +107,26 @@ def sorted(self):
         """
         return sorted(self)
 
-    def all(self, expression):
+    def all(self, expression=_NO_EXPR):
         """
         Returns True if all of the objects fulfill the expression. Returns False otherwise.
         """
-        return len(List(filter(expression, self))) == len(self)
+        for i in self:
```

Comment:
This is indeed better then a list comprehension, because it breaks sooner (as soon as 1 element is not fulfilling the expression). nice!

Reply:
This change is not for performance, it is for correctness.  The original code is not correct if the intent is to match the semantics of python's current any/all functions and their basis in the short circuiting behavior of and/or operators.  

https://stackoverflow.com/questions/17246388/does-alllist-use-short-circuit-evaluation/17246413#17246413